chat-components/src/components/notificationpane/NotificationPane.tsx

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import { ILabelStyles, Image, Label, Stack, Link, IStackStyles, IImageStyles, ILinkStyles } from "@fluentui/react";
+import { INotificationPaneInternal } from "./interfaces/common/INotificationPaneInternal";
+import DismissButton from "./subcomponents/DismissButton";
+import CloseChatButton from "./subcomponents/CloseChatButton";
+import { decodeComponentString } from "../../common/decodeComponentString";
+import { Ids } from "../../common/Constants";
+
+function NotificationPane(props: INotificationPaneInternal) {
+    const elementId = props.id ?? Ids.DefaultNotificationPaneId;
+
+    const containerStyles: IStackStyles = {
+        root: Object.assign({}, props.generalStyleProps)
+    };
+
+    const titleStyles: ILabelStyles = {
+        root: Object.assign({}, props.titleStyleProps)
+    };
+
+    const subtitleStyles: ILabelStyles = {
+        root: Object.assign({}, props.subtitleStyleProps)
+    };
+
+    const hyperlinkStyles: ILinkStyles = {
+        root: Object.assign({}, props.hyperlinkStyleProps)
+    };
+    const hyperlinkHoverStyles: ILinkStyles = {
+        root: Object.assign({}, props.hyperlinkHoverStyleProps)
+    };
+
+    const notificationIconProps = Object.assign({}, props.notificationIconProps);
+    const notificationIconStyles = Object.assign({}, props.notificationIconStyleProps);
+    const notificationIconImageStyles: IImageStyles = { root: {}, image: notificationIconStyles };
+    const notificationIconContainerStyles: IStackStyles = {
+        root: Object.assign({}, props.notificationIconContainerStyleProps)
+    };
+
+    const dismissButtonProps = Object.assign({}, props.dismissButtonProps);
+    const dismissButtonStyles = Object.assign({}, props.dismissButtonStyleProps);
+    const dismissButtonHoverStyles = Object.assign({}, props.dismissButtonHoverStyleProps);
+
+    const closeChatButtonProps = Object.assign({}, props.closeChatButtonProps);
```

Comment:
Do we need to re-assign all of these? Can't we directly use the props since we don't have any defaults?

Reply:
We do still need reassignment otherwise typescript complains the value can potentially be null.